### PR TITLE
feat(auth): adding oauth proxy deploy (AEROGEAR-8987)

### PR DESCRIPTION
**Pull request by laurafitzgerald**
## Motivation
https://issues.jboss.org/browse/AEROGEAR-8987

## What
Updating resources to add the oauth proxy deploy and configure so that it protects the service

## Why
Security is good

## How
See comments in [this JIRA](https://issues.jboss.org/browse/AEROGEAR-8371) about the manual steps

## Verification Steps
 
I've added verification steps below, but please keep in mind that this is a draft pr for feedback and the route needs to be manually added right now. I will update this later to be done automatically.
I've also removed the Readiness and Liveness Probes to allow the pods to come back up but these should be added back later. 

1. Update docker image to `quay.io/aerogear/mobile-security-service-operator:0.1.0-ag-8987`
2. Run `make create-all
4. Verify that the Service is protected by the proxy and that login/logout works correctly

## Checklist:

- [x] Code has been tested locally by PR requester
- [x] Changes have been successfully verified by another team member 

## Progress

- [x] Deployment Config
- [x] ServiceAccount
- [x] Route
- [x] Service

### CODE REVIEW

File: deploy/crds/mobile-security-service_v1alpha1_mobilesecurityservice_cr.yaml [hunk 1]
```diff
@@ -27,7 +30,7 @@ spec:
   # NOTE:
   # - Replace "{{clusterHost}}" for your cluster HOST. E.g 192.168.64.9
   # - You can use "$ oc status | grep server" or "kubectl cluster-info" to get this cluster info
-  clusterHost: "{{clusterHost}}"
+  clusterHost: "192.168.64.8"
```
Comment (camilamacedo86) [inline deploy/crds/mobile-security-service_v1alpha1_mobilesecurityservice_cr.yaml]: It should not be commit. Just to remember. It is boring and I hope that we will able to get this value automatically and remove this step of configuration.

File: pkg/controller/mobilesecurityservice/deployments.go [hunk 2]
```diff
@@ -38,51 +37,18 @@ func (r *ReconcileMobileSecurityService) buildAppDeployment(m *mobilesecurityser
...
 				},
 				Spec: corev1.PodSpec{
+					ServiceAccountName: "mobile-security-service-operator",
 					Containers: []corev1.Container{{
 						Image:           m.Spec.Image,
 						Name:            m.Spec.ContainerName,
 						ImagePullPolicy: corev1.PullAlways,
 						Ports: []corev1.ContainerPort{{
-							ContainerPort: m.Spec.Port,
+							ContainerPort: 3000,
```
Comment (camilamacedo86) [inline pkg/controller/mobilesecurityservice/deployments.go]: We should use the port from the env variables here since it is the app port. I know that it is not related to your PR.

File: pkg/controller/mobilesecurityservice/deployments.go [hunk 3]
```diff
@@ -38,51 +37,18 @@ func (r *ReconcileMobileSecurityService) buildAppDeployment(m *mobilesecurityser
...
 							Name:          "http",
 							Protocol:      "TCP",
 						}},
 						// Get the value from the ConfigMap
 						Env: *envinronment,
-						ReadinessProbe: &corev1.Probe{
```
Comment (camilamacedo86) [inline pkg/controller/mobilesecurityservice/deployments.go]: We cannot remove it. It will do the checks to ensure that the app and database are read and working well.

File: pkg/controller/mobilesecurityservice/deployments.go [hunk 4]
```diff
@@ -91,6 +57,30 @@ func (r *ReconcileMobileSecurityService) buildAppDeployment(m *mobilesecurityser
...
 						},
+					},{
+						Image:           "docker.io/openshift/oauth-proxy:v1.1.0",
+						Name:            "oauth-proxy",
+						Ports: []corev1.ContainerPort{{
+							ContainerPort: 4180,
+							Name:          "public",
+							Protocol:      "TCP",
+						}},
+						Args: []string{
```
Comment (camilamacedo86) [inline pkg/controller/mobilesecurityservice/deployments.go]: IMHO it could be in the helper as the func to get the Environment variables.

File: pkg/controller/mobilesecurityservice/deployments.go [hunk 5]
```diff
@@ -91,6 +57,30 @@ func (r *ReconcileMobileSecurityService) buildAppDeployment(m *mobilesecurityser
...
+							"--http-address=0.0.0.0:4180",
+							"--https-address=",
+							"--provider=openshift",
+							"--openshift-service-account=mobile-security-service-operator",
+							"--upstream=http://localhost:3000",
+							"--openshift-sar=[{\"namespace\":\"mobile-security-service-operator\",\"resource\":\"services\",\"name\":\"mobile-security-service-app\",\"verb\":\"get\"}]",
```
Comment (camilamacedo86) [inline pkg/controller/mobilesecurityservice/deployments.go]: This part probally should be changed to the m.name and m.namespac instead of be fixed the name of the service and namespace.

File: pkg/controller/mobilesecurityservice/deployments.go [hunk 6]
```diff
@@ -91,6 +57,30 @@ func (r *ReconcileMobileSecurityService) buildAppDeployment(m *mobilesecurityser
...
+							"--cookie-secure=true",
+							"--cookie-secret=SECRET",
```
Comment (camilamacedo86) [inline pkg/controller/mobilesecurityservice/deployments.go]: Have we secret? I don't think so. So, it shows not required.

File: pkg/controller/mobilesecurityservice/helpers.go [hunk 7]
```diff
@@ -44,8 +44,6 @@ func getAppEnvVarsMap(m *mobilesecurityservicev1alpha1.MobileSecurityService) ma
 		"PGHOST":                           m.Spec.DatabaseHost,
 		"LOG_LEVEL":                        m.Spec.LogLevel,
 		"LOG_FORMAT":                       m.Spec.LogFormat,
-		"ACCESS_CONTROL_ALLOW_ORIGIN":      m.Spec.AccessControlAllowOrigin,
```
Comment (camilamacedo86) [inline pkg/controller/mobilesecurityservice/helpers.go]: Are the environment variables removed from the Mobile Security Service project?

Comment (craicoverflow) [reply pkg/controller/mobilesecurityservice/helpers.go]: Yes, I am also wondering why it has been removed?

File: pkg/controller/mobilesecurityservice/deployments.go [hunk 8]
```diff
@@ -91,6 +57,30 @@ func (r *ReconcileMobileSecurityService) buildAppDeployment(m *mobilesecurityser
...
+							ContainerPort: 4180,
+							Name:          "public",
+							Protocol:      "TCP",
+						}},
+						Args: []string{
+							"--http-address=0.0.0.0:4180",
+							"--https-address=",
+							"--provider=openshift",
+							"--openshift-service-account=mobile-security-service-operator",
+							"--upstream=http://localhost:3000",
```
Comment (craicoverflow) [inline pkg/controller/mobilesecurityservice/deployments.go]: Should this be localhost:3000 or something else?

Comment (laurafitzgerald) [reply pkg/controller/mobilesecurityservice/deployments.go]: It seemed strange to me too but yes, it should be localhost:3000. More info here: https://github.com/openshift/oauth-proxy#using-this-proxy-with-openshift

**Comment by laurafitzgerald** [comment]
@camilamacedo86 @grdryn @craicoverflow @davidffrench taking this one of draft now, thanks to the Route changes by @camilamacedo86 I've now been able to complete the oauth configuaration, this is done as a default when the application is installed. I've updated the verification steps and this one is now ready for review and verification. pinging @grdryn as he is currently working on a similar change for ups.

File: deploy/service_account.yaml [hunk 9]
```diff
@@ -2,4 +2,7 @@ apiVersion: v1
 kind: ServiceAccount
 metadata:
   name: mobile-security-service-operator
-  namespace: mobile-security-service-operator
\ No newline at end of file
+  namespace: mobile-security-service-operator
+  annotations:
+    serviceaccounts.openshift.io/oauth-redirectreference.mobile-security-service-app: >-
+      {"kind":"OAuthRedirectReference","apiVersion":"v1","reference":{"kind":"Route","name":"service"}}
```
Comment (camilamacedo86) [inline deploy/service_account.yaml]: Could you add a comment here, and in the MSS CR on this attribute and a new topic under the COnfigurations one in the README to explain that here is the route name and it can be changed?Wdyt?

File: pkg/apis/mobilesecurityservice/v1alpha1/mobilesecurityservice_types.go [hunk 10]
```diff
@@ -29,17 +29,18 @@ type MobileSecurityServiceSpec struct {
...
+	MemoryLimit     string `json:"memoryLimit"`
+	MemoryRequest   string `json:"memoryRequest"`
 
 	//CR optional configuration values
-	ConfigMapName                 string `json:"configMapName,omitempty"`
-	RouteName                     string `json:"routeName,omitempty"`
-
+	ConfigMapName string `json:"configMapName,omitempty"`
+	RouteName     string `json:"routeName,omitempty"`
+	OAuth         bool   `json:"oAuth"`
```
Comment (camilamacedo86) [inline pkg/apis/mobilesecurityservice/v1alpha1/mobilesecurityservice_types.go]: These values are not optional. Could you just move them above of the comment `//CR optional configuration values`?

File: pkg/controller/mobilesecurityservice/deployments.go [hunk 11]
```diff
@@ -2,19 +2,105 @@ package mobilesecurityservice
...
+	"k8s.io/api/extensions/v1beta1"
+	"k8s.io/apimachinery/pkg/api/resource"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 )
 
 //buildDeployment returns the Deployment object using as image the MobileSecurityService App ( UI + REST API)
 func (r *ReconcileMobileSecurityService) buildDeployment(m *mobilesecurityservicev1alpha1.MobileSecurityService) *v1beta1.Deployment {
+
```
Comment (camilamacedo86) [inline pkg/controller/mobilesecurityservice/deployments.go]: Is it a new Deployment then it should be a new func for it. E.g buildOauthDeployment.

File: pkg/controller/mobilesecurityservice/helpers.go [hunk 12]
```diff
@@ -12,6 +12,14 @@ func getAppLabels(name string) map[string]string {
 	return map[string]string{"app": "mobilesecurityservice", "mobilesecurityservice_cr": name}
 }
 
+//getConfigMapName returns an string name with the name of the configMap
+func getConfigMapName(m *mobilesecurityservicev1alpha1.MobileSecurityService) string {
```
Comment (camilamacedo86) [inline pkg/controller/mobilesecurityservice/helpers.go]: Why it was added here? You can use the utils.GetConfigMapName(instance). This code impl was centralized.

File: pkg/controller/mobilesecurityservice/services.go [hunk 13]
```diff
@@ -2,16 +2,18 @@ package mobilesecurityservice
...
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 )
 
 //buildService returns the service resource
 func (r *ReconcileMobileSecurityService) buildService(m *mobilesecurityservicev1alpha1.MobileSecurityService) *corev1.Service {
 	ls := getAppLabels(m.Name)
+	var targetPort intstr.IntOrString
+	targetPort = intstr.FromInt(int(m.Spec.OAuthPort))
```
Comment (camilamacedo86) [inline pkg/controller/mobilesecurityservice/services.go]: Just a suggestion. You can replace. 

```go
 var targetPort intstr.IntOrString
	targetPort = intstr.FromInt(int(m.Spec.OAuthPort))
```
For 
```go
targetPort := intstr.FromInt(int(m.Spec.OAuthPort))
```

File: pkg/controller/mobilesecurityservice/deployments.go [hunk 14]
```diff
@@ -2,19 +2,105 @@ package mobilesecurityservice
...
+		}},
+		Args: []string{
+			"--http-address=0.0.0.0:4180",
+			"--https-address=",
+			"--provider=openshift",
+			"--openshift-service-account=mobile-security-service-operator",
+			"--upstream=http://localhost:3000",
+			"--openshift-sar=[{\"namespace\":\"mobile-security-service-operator\",\"resource\":\"services\",\"name\":\"mobile-security-service-app\",\"verb\":\"get\"}]",
+			"--cookie-secure=true",
+			"--cookie-secret=SECRET",
```
Comment (camilamacedo86) [inline pkg/controller/mobilesecurityservice/deployments.go]: Are we using Secret? I think that the Secret is not required.

File: pkg/controller/mobilesecurityservice/deployments.go [hunk 15]
```diff
@@ -2,19 +2,105 @@ package mobilesecurityservice
...
+			Name:          "public",
+			Protocol:      "TCP",
+		}},
+		Args: []string{
+			"--http-address=0.0.0.0:4180",
+			"--https-address=",
+			"--provider=openshift",
+			"--openshift-service-account=mobile-security-service-operator",
+			"--upstream=http://localhost:3000",
+			"--openshift-sar=[{\"namespace\":\"mobile-security-service-operator\",\"resource\":\"services\",\"name\":\"mobile-security-service-app\",\"verb\":\"get\"}]",
```
Comment (camilamacedo86) [inline pkg/controller/mobilesecurityservice/deployments.go]: The namespace cannot be fixed and it should be `instance.namespace`

Comment (camilamacedo86) [reply pkg/controller/mobilesecurityservice/deployments.go]: The name cannot be fixed and should be `instance.name` which will not be by default the value `mobile-security-service-app`

File: pkg/controller/mobilesecurityservice/deployments.go [hunk 16]
```diff
@@ -2,19 +2,105 @@ package mobilesecurityservice
...
+	var containers []corev1.Container
+	authContainer := corev1.Container{
+		Image: "docker.io/openshift/oauth-proxy:v1.1.0",
+		Name:  "oauth-proxy",
+		Ports: []corev1.ContainerPort{{
+			ContainerPort: m.Spec.OAuthPort,
+			Name:          "public",
+			Protocol:      "TCP",
+		}},
+		Args: []string{
```
Comment (camilamacedo86) [inline pkg/controller/mobilesecurityservice/deployments.go]: IHMO the args should be a func in the helper as for example getArgsForOauthDeployment(instance *...)

File: pkg/controller/mobilesecurityservice/deployments.go [hunk 17]
```diff
@@ -2,19 +2,105 @@ package mobilesecurityservice
...
+			"--cookie-httponly=false",
+			"--bypass-auth-for=/api/init",
+			"--bypass-auth-for=/api/healthz",
+			"--bypass-auth-for=/api/ping",
+			"--pass-user-headers=true",
+		},
+		TerminationMessagePath:   "/dev/termination-log",
+		TerminationMessagePolicy: "File",
+	}
+	containers = append(containers, authContainer)
```
Comment (camilamacedo86) [inline pkg/controller/mobilesecurityservice/deployments.go]: IMHO we need to encapsulate properly the code here. So we need to have func which will return the OauthContainer and another that will return the Deploy container. Then the deployment parameter which receives an array of containers will call both. Following an example. 

```go
func buildOauthContainer(...) corev1.Container{

}

func buildDeploymentContainer(...) corev1.Container{

}

{...}
//The following part will call the func which will return an []Containers

Spec: corev1.PodSpec{
					ServiceAccountName: "mobile-security-service-operator",
					Containers:         getDeploymentContainers(m),
				},

func getDeploymentContainers(..) []corev1.Container{
// Here we call the func whichs build the conatiners and create a array with 2 positions and add each one in one position then return it.

```

File: pkg/controller/mobilesecurityservice/deployments.go [hunk 18]
```diff
@@ -2,19 +2,105 @@ package mobilesecurityservice
...
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 )
 
 //buildDeployment returns the Deployment object using as image the MobileSecurityService App ( UI + REST API)
 func (r *ReconcileMobileSecurityService) buildDeployment(m *mobilesecurityservicev1alpha1.MobileSecurityService) *v1beta1.Deployment {
+
+	environment := buildAppEnvVars(m)
+	var containers []corev1.Container
+	authContainer := corev1.Container{
+		Image: "docker.io/openshift/oauth-proxy:v1.1.0",
```
Comment (camilamacedo86) [inline pkg/controller/mobilesecurityservice/deployments.go]: It could be in the spec.

File: pkg/apis/mobilesecurityservice/v1alpha1/mobilesecurityservice_types.go (hunk 10 above)
Comment (laurafitzgerald) [reply pkg/apis/mobilesecurityservice/v1alpha1/mobilesecurityservice_types.go]: @camilamacedo86 apologies, this should have been removed. It's not needed any more.

File: pkg/controller/mobilesecurityservice/helpers.go (hunk 12 above)
Comment (laurafitzgerald) [reply pkg/controller/mobilesecurityservice/helpers.go]: ah yes, I believe I introduced this in the rebase. Meant to remove. will do now.

File: pkg/controller/mobilesecurityservice/services.go (hunk 13 above)
Comment (laurafitzgerald) [reply pkg/controller/mobilesecurityservice/services.go]: 👍  will do

File: pkg/controller/mobilesecurityservice/deployments.go (hunk 15 above)
Comment (laurafitzgerald) [reply pkg/controller/mobilesecurityservice/deployments.go]: yes makes sense, I will investigate how I can parameterise these. and probably the service account as well?

File: pkg/controller/mobilesecurityservice/deployments.go (hunk 17 above)
Comment (laurafitzgerald) [reply pkg/controller/mobilesecurityservice/deployments.go]: @camilamacedo86 yes that looks cleaner. I will do that now.

**Comment by laurafitzgerald** [comment]
@camilamacedo86 thanks for the feedback, addressing the comments now.

File: pkg/controller/mobilesecurityservice/deployments.go (hunk 17 above)
Comment (laurafitzgerald) [reply pkg/controller/mobilesecurityservice/deployments.go]: Resolved in additional comment. Resolving conversation.  Please unresolve if further changes required. :)

File: pkg/controller/mobilesecurityservice/services.go (hunk 13 above)
Comment (laurafitzgerald) [reply pkg/controller/mobilesecurityservice/services.go]: Resolved in additional comment. Resolving conversation.  Please unresolve if further changes required. :)

File: pkg/controller/mobilesecurityservice/helpers.go (hunk 12 above)
Comment (laurafitzgerald) [reply pkg/controller/mobilesecurityservice/helpers.go]: Resolved in additional comment. Resolving conversation.  Please unresolve if further changes required. :)

File: pkg/apis/mobilesecurityservice/v1alpha1/mobilesecurityservice_types.go (hunk 10 above)
Comment (laurafitzgerald) [reply pkg/apis/mobilesecurityservice/v1alpha1/mobilesecurityservice_types.go]: Resolved in additional comment. Resolving conversation.  Please unresolve if further changes required. :)

File: pkg/controller/mobilesecurityservice/deployments.go (hunk 15 above)
Comment (camilamacedo86) [reply pkg/controller/mobilesecurityservice/deployments.go]: You can use instance.Namespace and instance.Name.

File: pkg/controller/mobilesecurityservice/deployments.go (hunk 16 above)
Comment (laurafitzgerald) [reply pkg/controller/mobilesecurityservice/deployments.go]: Resolved in additional commit. Resolving conversation.  Please unresolve if further changes required. :)

File: pkg/controller/mobilesecurityservice/deployments.go (hunk 15 above)
Comment (laurafitzgerald) [reply pkg/controller/mobilesecurityservice/deployments.go]: Resolved in additional commit. Resolving conversation.  Please unresolve if further changes required. :)

File: pkg/controller/mobilesecurityservice/deployments.go (hunk 14 above)
Comment (laurafitzgerald) [reply pkg/controller/mobilesecurityservice/deployments.go]: @camilamacedo86 without it the following error is produced:
`

2019/04/30 18:26:30 main.go:134: Invalid configuration:
--
  | missing setting: cookie-secret

``

File: README.adoc [hunk 19]
```diff
@@ -358,8 +357,8 @@ NOTE: The image/tag used from https://github.com/aerogear/mobile-security-servic
 
 |===
 | *Command*                     | *Description*
-| `make create-all`             | Create mobile-security-service-operator namespace, operator, service and roles.
-| `make delete-all`             | Delete mobile-security-service-operator namespace, operator, service and roles.
+| `make create-all`             | Create {namespace} namespace, operator, service and roles.
+| `make delete-all`             | Delete {namespace} namespace, operator, service and roles.
 | `make create-oper`            | Create mobile-security-service namespace, operator and roles.
```
Comment (camilamacedo86) [inline README.adoc]: shows that needs to be changed as well.
I'd recommend do a rebase with the master and change the make command as well to get the $Namespace var.

File: deploy/service_account.yaml (hunk 9 above)
Comment (laurafitzgerald) [reply deploy/service_account.yaml]: Resolved in additional commit. Resolving conversation. Please unresolve if further changes required. :)

File: README.adoc [hunk 20]
```diff
@@ -129,22 +130,20 @@ Use the following command to delete the Operator, the Service and its Database a
...
 
 NOTE:
 
-* All values used in the default configuration came from the config-map which is managed and created by the Operator. This config map will be created in the Operator namespace and its name is defined by the attribute `configMapName` in the link:./deploy/crds/mobile-security-service_v1alpha1_mobilesecurityservice_cr.yaml[MobileSecurityService CR].
-* If the name of this ConfigMap be not specified then the name of the Mobile Security Service instance will be used instead of.
+* All values used in the default configuration are sourced from the config-map which is managed and created by the Operator. This config map will be created in the Operator namespace and its name is defined by the attribute `configMapName` in the link:./deploy/crds/mobile-security-service_v1alpha1_mobilesecurityservice_cr.yaml[MobileSecurityService CR].
+* If the name of this ConfigMap is not specified then the name of the Mobile Security Service instance will be used instead.
 
-=== Role for no-privilege users are able to create the Mobile Security Service App, Database and/or Bind and Unbind their apps
+=== Configure the required roles
```
Comment (camilamacedo86) [inline README.adoc]: It is not required. It is optional.  it is an option to configure "Role for no-privilege users are able to create the Mobile Security Service App, Database and/or Bind and Unbind their apps"

File: README.adoc [hunk 21]
```diff
@@ -6,6 +6,7 @@ ifdef::env-github[]
 :caution-caption: :fire:
 :warning-caption: :warning:
 :table-caption!:
+:namespace: mobile-security-service
```
Comment (camilamacedo86) [inline README.adoc]: The namespace created by default when we run the make commands is : mobile-security-service-operator

File: README.adoc [hunk 22]
```diff
@@ -125,22 +126,20 @@ Use the following command to delete the Operator, the Service and its Database a
...
 
 NOTE:
 
-* All values used in the default configuration came from the config-map which is managed and created by the Operator. This config map will be created in the Operator namespace and its name is defined by the attribute `configMapName` in the link:./deploy/crds/mobile-security-service_v1alpha1_mobilesecurityservice_cr.yaml[MobileSecurityService CR].
-* If the name of this ConfigMap be not specified then the name of the Mobile Security Service instance will be used instead of.
+* All values used in the default configuration are sourced from the config-map which is managed and created by the Operator. This config map will be created in the Operator namespace and its name is defined by the attribute `configMapName` in the link:./deploy/crds/mobile-security-service_v1alpha1_mobilesecurityservice_cr.yaml[MobileSecurityService CR].
+* If the name of this ConfigMap is not specified then the name of the Mobile Security Service instance will be used instead.
 
-=== Role for no-privilege users are able to create the Mobile Security Service App, Database and/or Bind and Unbind their apps
+=== Configure roles for no-priveledge users (optional)
```
Comment (camilamacedo86) [inline README.adoc]: I think the (optional) is not required since the topic is Configuration and options

File: README.adoc [hunk 23]
```diff
@@ -125,22 +126,20 @@ Use the following command to delete the Operator, the Service and its Database a
 $ make delete-all
 ----
 
-IMPORTANT: To create all you need be logged as admin. Use `oc login -u system:admin`.
-
-== Configurations and Options
+== Configuration and Options
```
Comment (camilamacedo86) [inline README.adoc]: Shows that we could add here a sub-topic to explain the OAuth setup and for example if the routeName in CR be changed that is required change it in thee service_account.yaml file which is the config of the service-account used by the operator.

**Comment by camilamacedo86** [comment]
Hi @laurafitzgerald,

You made a fantastic work here. The code shows perfect fine. Just a few comments regard the readme. 
I will test/verify your change now.

PS.: I'd like to suggest do not create tag images for the test in <quay.io/aerogear/mobile-security-service-operator> because it will be there. We can remove but is required to remember to do that. I think is better we use our personal org. E.g  docker.io/cmacedo/mobile-security-service-operator:fix-db.

File: pkg/controller/mobilesecurityservice/route.go [hunk 24]
```diff
@@ -3,13 +3,16 @@ package mobilesecurityservice
...
 	routev1 "github.com/openshift/api/route/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 )
 
 //buildRoute returns the route resource
 func (r *ReconcileMobileSecurityService) buildRoute(m *mobilesecurityservicev1alpha1.MobileSecurityService) *routev1.Route {
+
+	web := int32(100)
```
Comment (davidffrench) [inline pkg/controller/mobilesecurityservice/route.go]: @laurafitzgerald Could you add a comment to this. It is unclear to me why this variable is called web with a default value and passed into the Weight property.

Comment (laurafitzgerald) [reply pkg/controller/mobilesecurityservice/route.go]: @davidffrench nicely spotted. I would say that I introduced that name incorrectly. The targetport value is "web" but this variable relates to the [weight value](https://docs.openshift.com/container-platform/3.11/architecture/networking/routes.html#alternateBackends). Would updating the variable name to weight be sufficient?

Comment (laurafitzgerald) [reply pkg/controller/mobilesecurityservice/route.go]: As discussed, resolved by removing the weight value as it defaults and there is only one service so we don't need to specify the weight.
